node/packages/client/src/cosmos/parser/__tests__/cosmos.test.ts

```diff
@@ -45,29 +63,217 @@ describe('parseTx', () => {
     const address = 'cosmos14e25lpsedq863vgweqg4m9n0z28c203kfdlzmz'
 
     const expected: Tx = {
+      address: 'cosmos14e25lpsedq863vgweqg4m9n0z28c203kfdlzmz',
+      caip2: 'cosmos:cosmoshub-4',
       txid: tx.txid,
       blockHash: tx.blockHash,
       blockHeight: tx.blockHeight,
       blockTime: tx.timestamp,
       confirmations: tx.confirmations,
       status: Status.Confirmed,
-      address: address,
-      caip2: 'cosmos:cosmoshub-4',
-      value: tx.value,
       transfers: [
         {
           type: TransferType.Receive,
-          from: 'cosmos1t5u0jfg3ljsjrh2m9e47d4ny2hea7eehxrzdgd',
-          to: address,
           caip19: 'cosmos:cosmoshub-4/slip44:118',
+          from: 'cosmos1t5u0jfg3ljsjrh2m9e47d4ny2hea7eehxrzdgd',
+          to: 'cosmos14e25lpsedq863vgweqg4m9n0z28c203kfdlzmz',
           totalValue: '2002965',
-          components: [{ value: '2002965' }],
+          components: [
+            {
+              value: '2002965',
+            },
+          ],
         },
       ],
+      value: '2002965',
+      data: {
+        parser: 'cosmos',
+        method: 'send',
+        extras: {
+          from: 'cosmos1t5u0jfg3ljsjrh2m9e47d4ny2hea7eehxrzdgd',
+          to: 'cosmos14e25lpsedq863vgweqg4m9n0z28c203kfdlzmz',
+          caip19: 'cosmos:cosmoshub-4/slip44:118',
+          value: '2002965',
+        },
+      },
     }
 
     const actual = await txParser.parse(tx, address)
 
     expect(expected).toEqual(actual)
   })
+
+  it('should be able to parse a withdraw_delegator_reward tx', async () => {
+    const { tx } = reward
+    const address = 'cosmos179k2lz70rxvjrvvr65cynw9x5c8v3kftg46v05'
+
+    const expected: Tx = {
+      address: 'cosmos179k2lz70rxvjrvvr65cynw9x5c8v3kftg46v05',
+      blockHeight: 9636957,
+      blockTime: 1645207449,
+      caip2: 'cosmos:cosmoshub-4',
+      confirmations: 358801,
+      status: Status.Confirmed,
+      transfers: [
+        {
+          type: TransferType.Receive,
+          caip19: 'cosmos:cosmoshub-4/slip44:118',
+          from: 'cosmosvaloper1hdrlqvyjfy5sdrseecjrutyws9khtxxaux62l7',
+          to: 'cosmos179k2lz70rxvjrvvr65cynw9x5c8v3kftg46v05',
+          totalValue: '39447',
+          components: [
+            {
+              value: '39447',
+            },
+          ],
+        },
+      ],
+      txid: 'E34AFB3A28198957040073034E16D4A979B403E672859651B41C207538136ABE',
+      value: '39447',
+      data: {
+        parser: 'cosmos',
+        method: 'withdraw_delegator_reward',
+        extras: {
+          from: 'cosmosvaloper1hdrlqvyjfy5sdrseecjrutyws9khtxxaux62l7',
```

Comment:
Just to be 100% sure, as a user I should never need to interact directly with `cosmosvalcons` and `cosmosvaloper` addresses right?
Currently in web, we have a validation that only allows sending to `cosmos` HRP and I'm wondering if we should support these two also.